src/test/java/core/basesyntax/service/impl/ReportGeneratorImplTest.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ReportGeneratorImplTest {
+    private static final String COMA_SEPARATOR = ",";
+    private static final String TITLE = "fruit,quantity";
+    private Storage storage;
+    private ReportGenerator reportGenerator;
+
+    @BeforeEach
+    void setUp() {
+        storage = new Storage();
+        reportGenerator = new ReportGeneratorImpl(storage);
+    }
+
+    @AfterEach
+    void tearDown() {
+        storage.clear();
+    }
+
+    @Test
+    void generate_dataIsValid_isOk() {
+        storage.addFruit("apple", 5);
+        storage.addFruit("banana", 10);
+        List<String> formattedList = new ArrayList<>();
+        formattedList.add(TITLE);
+        storage.getFruits().forEach((fruit, quantity) -> formattedList
+                .add(fruit + COMA_SEPARATOR + quantity));
+        List<String> actual = formattedList;
+        List<String> expected = Arrays.asList(
+                new String("fruit,quantity"),
+                new String("banana,10"),
+                new String("apple,5"));
+        assertEquals(expected, actual);
+
+    }
+
+    @Test
+    void generate_emptyData_isOk() {
+        List<String> actual = reportGenerator.generate();
+        List<String> expected = Arrays.asList(new String("fruit,quantity"));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void generate_oneFruitData_isOk() {
+        storage.addFruit("apple", 5);
+        List<String> formattedList = new ArrayList<>();
+        formattedList.add(TITLE);
+        storage.getFruits().forEach((fruit, quantity) -> formattedList
+                .add(fruit + COMA_SEPARATOR + quantity));
+        List<String> actual = formattedList;
+        List<String> expected = Arrays.asList(
+                new String("fruit,quantity"),
+                new String("apple,5"));
+        assertEquals(expected, actual);

```

Comment:
and here too, just fill the storage with some test data, call report creation method and assert it's return value with some expected one
cases that would add for it:
* empty storage
* single fruit in the storage 
* several fruits in the storage
* a fruit with 0 quantity in the storage
* maybe even a fruit with negative quantity in the storage, but that depends on you implementation and if such situation is possible in it

Reply:
not resolved

Reply:
Sorry!!! I've done it , except  a fruit with negative quantity in the storage , because  FruitServiceImpl check it.